### PR TITLE
Update ObservationResponse

### PR DESCRIPTION
**Pull request by chejennifer**
- return ObsCount, EarliestDate, and LatestDate for each FacetObservation in ObservationResponse

Cases:
- For a series, the 3 fields are about the observations for that facet
- For a collection where the entity is "", obsCount is the number of entities with observations for this facet, EarliestDate and LatestDate are the earliest and latest dates that any entity has observations for this facet.
- For a collection with entities, the 3 fields are about the observation for that facet (i.e., obsCount will be 1, earliest and latest dates will be the same as the date of the returned observation)

Note: 
- for getting facet for a collection, if using the ObsCollectionPlaceVariableFacet cache, there is no earliestDate or obsCount information

### CODE REVIEW

File: internal/server/v2/facet/contained_in.go [hunk 1]
```diff
@@ -103,6 +103,8 @@ func ContainedInFacet(
 						continue
 					}
 					seenFacets[facetID] = struct{}{}
+					// TODO: Add additional ObsCount, EarliestDate, LatestDate information.
```
Comment (shifucun) [inline internal/server/v2/facet/contained_in.go:106]: Can you file a bug for this? @n-h-diaz can you help add this in Flume?

File: internal/server/v2/facet/contained_in.go [hunk 2]
```diff
@@ -134,7 +136,8 @@ func ContainedInFacet(
 					return nil, err
 				}
 				for _, entityData := range resp.ByVariable {
-					seenFacetId := map[string]struct{}{}
+					seenFacet := map[string]*pbv2.FacetObservation{}
+					facetIdOrder := []string{}
```
Comment (shifucun) [inline internal/server/v2/facet/contained_in.go:140]: nit: orderedFacetId

File: internal/server/v2/facet/contained_in.go [hunk 3]
```diff
@@ -143,19 +146,42 @@ func ContainedInFacet(
 					for _, entity := range childPlaces {
 						if facetData, ok := entityData.ByEntity[entity]; ok {
 							for _, item := range facetData.OrderedFacets {
-								if _, ok := seenFacetId[item.FacetId]; ok {
-									continue
+								// obsCount is the number of entities with data for this facet
+								obsCount := int32(1)
```
Comment (shifucun) [inline internal/server/v2/facet/contained_in.go:150]: obsCount := 1 should work

File: internal/server/v2/facet/contained_in.go [hunk 4]
```diff
@@ -143,19 +146,42 @@ func ContainedInFacet(
...
-								mergedFacetData.OrderedFacets = append(
-									mergedFacetData.OrderedFacets,
-									&pbv2.FacetObservation{
-										FacetId: item.FacetId,
-									},
-								)
 							}
 						}
 					}
+					for _, facetId := range facetIdOrder {
```
Comment (shifucun) [inline internal/server/v2/facet/contained_in.go:179]: do you want to sort the facetIdOrder here?

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json [hunk 5]
```diff
@@ -1,89 +0,0 @@
-{
```
Comment (shifucun) [inline internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json:1]: why is this file deleted?

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json [hunk 6]
```diff
@@ -5,25 +5,45 @@
...
-              "facet_id": "1145703171"
+              "facet_id": "2176550201",
+              "obs_count": 58,
+              "earliest_date": "1970",
+              "latest_date": "2021"
             },
             {
-              "facet_id": "1541763368"
+              "facet_id": "1145703171",
+              "obs_count": 58,
```
Comment (shifucun) [inline internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json:15]: why there is no earliest_date?

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_latest.json [hunk 7]
```diff
@@ -1,105 +0,0 @@
-{
```
Comment (shifucun) [inline internal/server/v2/facet/golden/contained_in_facet/CA_County_latest.json:1]: why deleted?

File: internal/server/v2/facet/golden/series_facet/series_facet.json [hunk 8]
```diff
@@ -4,205 +4,164 @@
       "by_entity": {
         "Earth": {
           "ordered_facets": [
+            {
+              "facet_id": "3981252704",
```
Comment (shifucun) [inline internal/server/v2/facet/golden/series_facet/series_facet.json:8]: why is there new data here?

File: internal/server/v2/facet/series.go [hunk 9]
```diff
@@ -64,25 +64,55 @@ func SeriesFacet(
...
+				if _, ok := varEntityFacets[variable][entity]; !ok {
+					varEntityFacets[variable][entity] = map[string]*pb.PlaceVariableFacet{}
+				}
+				facetList := row.Data.(*pb.PlaceVariableFacets).GetPlaceVariableFacets()
+				for _, placeVarFacet := range facetList {
+					facetID := util.GetFacetID(placeVarFacet.Facet)
+					seenPlaceVarFacet, ok := varEntityFacets[variable][entity][facetID]
+					// If we've seen this facet already and it has the EarliestDate field,
+					// don't override the mapped facet info.
+					if ok && seenPlaceVarFacet.EarliestDate != "" {
```
Comment (shifucun) [inline internal/server/v2/facet/series.go:84]: why only checking EarliestDate?

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json (hunk 5 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json:1]: it's not used in any of the tests, I think it's outdated

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_latest.json (hunk 7 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_latest.json:1]: also not used in any tests

File: internal/server/v2/facet/golden/series_facet/series_facet.json (hunk 8 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/series_facet/series_facet.json:8]: originally, series facet was only getting facets from one of the BT versions (which was a bug)

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json (hunk 6 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json:15]: I'm actually not sure, but some of the facet information from the cache returns no earliest_date ..

File: internal/server/v2/facet/series.go (hunk 9 above)
Comment (chejennifer) [reply internal/server/v2/facet/series.go:84]: there were cases where multiple BT versions will return the same facet, but only one of the versions has the earliest date field populated, so we want to keep the one with the earliest date field

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json (hunk 6 above)
Comment (shifucun) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json:15]: I see, it was removed in https://github.com/datacommonsorg/mixer/pull/1295/files#diff-f92f6edcac465aee9633f109b36078f74067ded3b7f3fe7210be800fc60727a5. Can you add it back here?

File: internal/server/v2/facet/contained_in.go (hunk 3 above)
Comment (chejennifer) [reply internal/server/v2/facet/contained_in.go:150]: it does not because FacetObservation requires ObsCount to be int32

File: internal/server/v2/facet/series.go (hunk 9 above)
Comment (shifucun) [reply internal/server/v2/facet/series.go:84]: That seems to be a bug in Flume..

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json (hunk 5 above)
Comment (shifucun) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json:1]: I see, it was removed in https://github.com/datacommonsorg/mixer/pull/1295/files#diff-f92f6edcac465aee9633f109b36078f74067ded3b7f3fe7210be800fc60727a5. Can you add it back here?

File: internal/server/v2/facet/contained_in.go (hunk 2 above)
Comment (chejennifer) [reply internal/server/v2/facet/contained_in.go]: done

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json (hunk 5 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json:1]: oh, did you remove it for a reason? or by accident?

Comment (shifucun) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_2015.json:1]: I can not remember, likely by accident.

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json (hunk 6 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json:15]: Actually, just checked, it's because infrequent table version is from before the earliest_date change. Next data release will fix this, which maybe I can do one this week (just the data release)

Comment (shifucun) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json:15]: I see. After that, maybe update the logic that relies on earliest_date only.

File: internal/server/v2/facet/contained_in.go (hunk 4 above)
Comment (shifucun) [reply internal/server/v2/facet/contained_in.go]: Got it, then this is fine.

**Review by shifucun — APPROVED** [review]

File: internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json (hunk 6 above)
Comment (chejennifer) [reply internal/server/v2/facet/golden/contained_in_facet/CA_County_all.json:15]: added a todo

**Comment by chejennifer** [comment]
thanks for the review!

File: internal/server/v2/facet/contained_in.go (hunk 1 above)
Comment (chejennifer) [reply internal/server/v2/facet/contained_in.go:106]: filed bug: https://buganizer.corp.google.com/issues/311225915
